broadcast-message.php

```diff
@@ -49,23 +54,35 @@
    }
    else
    {
+      $wanted_domains = array_intersect($allowed_domains, $_POST['domains']);
+
       $table_mailbox = table_by_key('mailbox');
-	  $table_alias = table_by_key('alias');
-      
-	  $q = "select username from $table_mailbox union select goto from $table_alias " .
-		   "where goto not in (select username from $table_mailbox)";
-
-      $result = db_query ($q);
-      if ($result['rows'] > 0)
-      {
+      $table_alias = table_by_key('alias');
+
+      $recipients = [];
+
+      $q = "SELECT username from $table_mailbox WHERE ".db_in_clause("domain", $wanted_domains);
+      if(!isset($_POST['mailboxes_only']) || $_POST['mailboxes_only'] != "on") {
```

Comment:
With the checkbox value changed to "1", you can simplify this condition:

        if (intval(safepost('mailboxes_only')) == 0) {

BTW: a space after `if` would be nice.